language/compiler/ir_to_bytecode/syntax/src/syntax.lalrpop

```diff
@@ -343,12 +345,58 @@ FunctionBlock: (Vec<(Var_, Type)>, Block) = {
     "{" <locals: Declarations> <stmts: Statements> "}" => (locals, Block::new(stmts))
 }
 
+KindConstraint: KindConstraint = {
+    <n: Name> => KindConstraint::new(n, Kind::All),
+    <n: Name> ":" "all" => KindConstraint::new(n, Kind::All),
+    <n: Name> ":" "resource" => KindConstraint::new(n, Kind::Resource),
+    <n: Name> ":" "unrestricted" => KindConstraint::new(n, Kind::Unrestricted),
+}
+
+KindConstraintsImpl: KindConstraints = {
+    "<" <constraints: Comma<KindConstraint>> ">" => KindConstraints::new(constraints)
+}
+
+KindConstraints: KindConstraints = {
+    <constraints: KindConstraintsImpl?> => {
+        match constraints {
+            Some(constraints) => constraints,
+            None => KindConstraints::new(vec![]),
+        }
+    }
+}
+
+TypeParametersImpl: Vec<Type> = {
+    "<" <tys: Comma<Type>> ">" => tys
+}
+
+TypeParametersColonImpl: Vec<Type> = {
+    "::" <tys: TypeParametersImpl> => tys
```

Comment:
I thought we had discussed not doing this. And instead updating the lexing of identifiers?